lib/cmps/onetrust.ts

```diff
@@ -0,0 +1,42 @@
+import AutoConsentBase from "./base";
+import { TabActor } from "../types";
+
+export default class Onetrust extends AutoConsentBase {
+
+  prehideSelectors = ["#onetrust-banner-sdk,#onetrust-consent-sdk,.optanon-alert-box-wrapper,.onetrust-pc-dark-filter,.js-consent-banner"]
+
+  constructor() {
+    super("Onetrust");
+  }
+
+  detectCmp(tab: TabActor) {
+    return tab.elementExists("#onetrust-banner-sdk,.optanon-alert-box-wrapper");
+  }
+
+  detectPopup(tab: TabActor) {
+    return tab.elementsAreVisible("#onetrust-banner-sdk,.optanon-alert-box-wrapper");
+  }
+
+  async optOut(tab: TabActor) {
+    let res = true;
```

Comment:
Can you explain the need for this result chaining? Do we silently skip these actions if a previous step fails? Perhaps a helper function (in a similar style to [`success`](https://github.com/duckduckgo/autoconsent/blob/cdd6a3b792e827f769028f9fe603c88a09407d8a/lib/cmps/base.ts#L20)) could make this more readable?

Reply:
I tried to mimic the declarative rules processing. [There](https://github.com/duckduckgo/autoconsent/blob/muodov/js-onetrust/lib/cmps/base.ts#L148) we abort the whole process as soon as one step fails. This makes sense, because we don't want to click unrelated stuff accidentally. So this `res &&` was a way to cancel subsequent steps.
I think I can use `success()` here, thanks for the tip!